plugins/filter_aws/aws.c

```diff
@@ -427,8 +429,113 @@ static int get_ec2_tag_values(struct flb_filter_aws *ctx)
     return 0;
 }
 
+static int tag_is_present_in_list(struct flb_filter_aws *ctx, flb_sds_t tag,
+        const char *tags)
+{
+    const char *tag_start;
+    const char *tag_end;
+    const char *tag_sep_before;
+    const char *tags_start;
+    const char *tag_sep_after;
+    const char *tags_end;
+
+    tag_start = strstr(tags, tag);
+    if (!tag_start) {
+        return FLB_FALSE;
+    }
+    tag_end = tag_start + strlen(tag) - 1;
+
+    /* if substring exists it not necessarily implies that tag is included */
+    /* tag name must be exactly equal to the value between separators ',' */
+    /* f.e. without additional conditions, substring would accept Name for a,Namee,b */
+
+    tags_start = tags;
+    tags_end = tags + strlen(tags) - 1;
+
+    /* calculate pointers to characters directly before/after tag */
+    /* if tag is in the middle, then these pointers should be ',' */
+    tag_sep_before = (char*)(tag_start - 1);
+    tag_sep_after = (char*)(tag_end + 1);
+    if (tag_sep_before < tags_start) {
+        tag_sep_before = NULL;
+    }
+    if (tag_sep_after > tags_end) {
+        tag_sep_after = NULL;
+    }
```

Comment:
won't these ifs always run? Since the lines before

Reply:
No, these may run up to twice for all tags.

These will only run when we check the first/last tag in the tags list. Otherwise these pointers will point to the valid memory and we will have to confirm if the value is semicolon.

Reply:
I guess the confusion might be because of the fact that `tags_start` and `tag_start` are two different variables. Same for _end ones.

Reply:
This is an old implementation, which no longer exists, so I am going to resolve the conversation.